src/checkers/inference/solver/frontend/TwoQualifiersLattice.java

```diff
@@ -0,0 +1,54 @@
+package checkers.inference.solver.frontend;
+
+import org.checkerframework.javacutil.AnnotationUtils;
+
+import java.util.Set;
+
+import javax.lang.model.element.AnnotationMirror;
+
+/**
+ * Special Lattice class for two qualifier type system.
+ * 
+ * @author jianchu
+ *
+ */
+public class TwoQualifiersLattice extends Lattice {
+
+    protected Set<AnnotationMirror> allTypes;
```

Comment:
Um... I see the reason. This is because in `TwoTypeLattice` you manually add `top` and `bottom` into the `allTypes`, which will failed if you add them into a set type of `Set<? extends AM>`.

Could you refer the implementation in `QualifierHeirarchy#getTypeQualifiers()`? To be specific, you can get rid of this problem by:

```java
Set<AM> tempSet;
tempSet.add(top and btm);
allTypes = Collections.unmodifiableSet(tempSet);